### PR TITLE
AWS-MSK implementation updates

### PR DESCRIPTION
**Pull request by bpathak-ons**
### What

* Enable app to use Kafka TLS / AWS-MSK
* Add the relevant AWS-MSK notes, configs and code

### How to review

* check if code changes make sense
* CI pass for build, audit and unit

### Who can review

* anyone

### CODE REVIEW

File: service/initialise.go [hunk 1]
```diff
@@ -94,7 +94,19 @@ func (e *ExternalServiceList) GetKafkaConsumer(ctx context.Context, cfg *config.
 // DoGetKafkaConsumer returns a Kafka Consumer group
 func (e *Init) DoGetKafkaConsumer(ctx context.Context, cfg *config.Config) (dpkafka.IConsumerGroup, error) {
 	cgChannels := dpkafka.CreateConsumerGroupChannels(1)
-	kafkaOffset := dpkafka.OffsetNewest
+
+	pConfig := &dpkafka.ProducerConfig{
+		KafkaVersion: &cfg.KafkaVersion,
+	}
+	if cfg.KafkaSecProtocol == "TLS" {
```
Comment (justinpjose) [inline service/initialise.go:101]: The value "TLS" should be a constant which should be ideally mentioned in the `config.go`. Use this as an example: https://github.com/ONSdigital/dp-observation-extractor/blob/develop/config/config.go#L13

**Review by justinpjose — COMMENTED** [review]
Can you please also doing the following:

- Update the headings in the README to be consistent
- Update config_test.go to check the default values of the kafka config
- Add tests to validate kafka configs
An example of all of the above can be found in this PR - https://github.com/ONSdigital/dp-observation-extractor/pull/61/files

Also,
- Update all the dependencies of the app (go get -u ./... and then go mod tidy )
- Update the go version from 1.17.1 to 1.17.2 in ci package

File: README.md [hunk 2]
```diff
@@ -47,8 +47,13 @@ An example event can be created using the helper script, `make produce`.
 | HEALTHCHECK_INTERVAL         | 30s                               | Time between self-healthchecks (`time.Duration` format)
 | HEALTHCHECK_CRITICAL_TIMEOUT | 90s                               | Time to wait until an unhealthy dependent propagates its state to make this app unhealthy (`time.Duration` format)
 | KAFKA_ADDR                   | "localhost:9092"                  | The address of Kafka (accepts list)
-| KAFKA_OFFSET_OLDEST          | true                              | Start processing Kafka messages in order from the oldest in the queue
-| KAFKA_NUM_WORKERS            | 1                                 | The maximum number of parallel kafka consumers
+| KAFKA_VERSION                | `1.0.2`                  | The version of Kafka
+| KAFKA_CONSUMER_WORKERS       | 1                        | The maximum number of parallel kafka consumers
```
Comment (gedge) [inline README.md:51]: where is this in config.go (I think you renamed it, from `KAFKA_NUM_WORKERS` for some reason)?

File: service/initialise.go [hunk 3]
```diff
@@ -94,7 +94,19 @@ func (e *ExternalServiceList) GetKafkaConsumer(ctx context.Context, cfg *config.
 // DoGetKafkaConsumer returns a Kafka Consumer group
 func (e *Init) DoGetKafkaConsumer(ctx context.Context, cfg *config.Config) (dpkafka.IConsumerGroup, error) {
 	cgChannels := dpkafka.CreateConsumerGroupChannels(1)
-	kafkaOffset := dpkafka.OffsetNewest
+
+	pConfig := &dpkafka.ProducerConfig{
```
Comment (gedge) [inline service/initialise.go:98]: this is a producer config, and `pConfig` is not used in this function
I think you need `ConsumerGroupConfig` here (perhaps called `cConfig`) and to use that in the below

File: config/config.go [hunk 4]
```diff
@@ -12,10 +12,15 @@ type Config struct {
 	GracefulShutdownTimeout    time.Duration `envconfig:"GRACEFUL_SHUTDOWN_TIMEOUT"`
 	HealthCheckInterval        time.Duration `envconfig:"HEALTHCHECK_INTERVAL"`
 	HealthCheckCriticalTimeout time.Duration `envconfig:"HEALTHCHECK_CRITICAL_TIMEOUT"`
-	KafkaAddr                  []string      `envconfig:"KAFKA_ADDR"                     json:"-"`
+	KafkaAddr                  []string      `envconfig:"KAFKA_ADDR"                   json:"-"`
```
Comment (gedge) [inline config/config.go:15]:
```suggestion
	KafkaAddr                  []string      `envconfig:"KAFKA_ADDR"`
```
no need to hide this value

File: service/initialise.go [hunk 5]
```diff
@@ -94,6 +97,18 @@ func (e *ExternalServiceList) GetKafkaConsumer(ctx context.Context, cfg *config.
 // DoGetKafkaConsumer returns a Kafka Consumer group
 func (e *Init) DoGetKafkaConsumer(ctx context.Context, cfg *config.Config) (dpkafka.IConsumerGroup, error) {
 	cgChannels := dpkafka.CreateConsumerGroupChannels(1)
+
+	cConfig := &dpkafka.ConsumerGroupConfig{
```
Comment (gedge) [inline service/initialise.go:101]: `cConfig` needs to be used in `NewConsumerGroup` call, below

File: service/initialise.go [hunk 6]
```diff
@@ -94,6 +97,18 @@ func (e *ExternalServiceList) GetKafkaConsumer(ctx context.Context, cfg *config.
...
+		KafkaVersion: &cfg.KafkaVersion,
```
Comment (gedge) [inline service/initialise.go:102]:
```suggestion
		KafkaVersion: &cfg.KafkaVersion,
		Offset:       &kafkaOffset,
```

**Pull request merged**